hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieFileGroupReader.java

```diff
@@ -105,7 +103,7 @@ public HoodieFileGroupReader(HoodieReaderContext<T> readerContext,
     HoodieTableConfig tableConfig = hoodieTableMetaClient.getTableConfig();
     this.recordMerger = readerContext.getRecordMerger(tableConfig.getRecordMergerStrategy());
     readerContext.setRecordMerger(this.recordMerger);
```

Comment:
Could you file a follow-up ticket to refactor the reader context construction?  It's still an anti-pattern to allow any setters which make it mutable and not safe.